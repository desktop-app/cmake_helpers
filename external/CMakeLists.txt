# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

macro(add_checked_subdirectory name)
    if (NOT DEFINED desktop_app_skip_libs
        OR NOT ${name} IN_LIST desktop_app_skip_libs)
        add_subdirectory(${name})
    endif()
endmacro()

add_checked_subdirectory(auto_updates)
add_checked_subdirectory(crash_reports)
if (LINUX AND NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
    add_checked_subdirectory(dbusmenu_qt)
endif()
add_checked_subdirectory(expected)
if (LINUX AND NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
    add_checked_subdirectory(fcitx_qt5)
    add_checked_subdirectory(fcitx5_qt5)
    add_checked_subdirectory(fcitx5_qt5_dbusaddons)
endif()
add_checked_subdirectory(ffmpeg)
add_checked_subdirectory(gsl)
if (LINUX AND NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
    add_checked_subdirectory(hime_im_client)
    add_checked_subdirectory(hime_qt)
endif()
if (add_hunspell_library)
	add_checked_subdirectory(hunspell)
endif()
add_checked_subdirectory(iconv)
if (LINUX AND NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
    add_checked_subdirectory(lxqt_qtplugin)
endif()
add_checked_subdirectory(lz4)
if (LINUX)
    add_checked_subdirectory(materialdecoration)
endif()
add_checked_subdirectory(minizip)
if (LINUX)
    add_checked_subdirectory(nimf_qt5)
endif()
add_checked_subdirectory(openal)
add_checked_subdirectory(openssl)
add_checked_subdirectory(opus)
add_checked_subdirectory(qt)
if (LINUX)
    add_checked_subdirectory(qt5ct)
    add_checked_subdirectory(qt5ct_qtplugin)
    add_checked_subdirectory(qt5ct_style)
endif()
add_checked_subdirectory(qr_code_generator)
add_checked_subdirectory(ranges)
add_checked_subdirectory(rlottie)
if (APPLE)
    add_checked_subdirectory(sp_media_key_tap)
endif()
if (LINUX AND NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
    add_checked_subdirectory(statusnotifieritem)
endif()
add_checked_subdirectory(ton)
add_checked_subdirectory(variant)
if (LINUX)
    add_checked_subdirectory(xdgiconloader)
endif()
add_checked_subdirectory(xxhash)
add_checked_subdirectory(zlib)
