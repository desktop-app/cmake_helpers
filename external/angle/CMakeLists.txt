# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

add_library(external_angle INTERFACE IMPORTED GLOBAL)
add_library(desktop-app::external_angle ALIAS external_angle)

if (WIN32)
    target_include_directories(external_angle SYSTEM
    INTERFACE
        ${libs_loc}/tg_angle/include
    )
    target_link_libraries(external_angle
    INTERFACE
        ${libs_loc}/tg_angle/out/$<IF:$<CONFIG:Debug>,Debug,Release>/tg_angle.lib
        dxguid.lib
    )
    target_link_libraries(external_angle
    INTERFACE
        desktop-app::win_directx_helper
    )
endif()
