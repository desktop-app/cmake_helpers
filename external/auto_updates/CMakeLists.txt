# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

add_library(external_auto_updates INTERFACE IMPORTED GLOBAL)
add_library(desktop-app::external_auto_updates ALIAS external_auto_updates)

if (WIN32 AND NOT DESKTOP_APP_USE_PACKAGED)
    add_subdirectory(lzma)
    target_link_libraries(external_auto_updates
    INTERFACE
        desktop-app::external_lzma
    )
else()
    add_subdirectory(xz)
    target_link_libraries(external_auto_updates
    INTERFACE
        desktop-app::external_xz
    )
endif()
