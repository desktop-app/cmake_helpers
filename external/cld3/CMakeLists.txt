# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

add_library(external_cld3 INTERFACE IMPORTED GLOBAL)
add_library(desktop-app::external_cld3 ALIAS external_cld3)

add_library(external_cld3_bundled STATIC)
init_target(external_cld3_bundled "(external)")

set(cld3_loc ${third_party_loc}/cld3)
set(cld3_src ${cld3_loc}/src)

set(gen_loc ${CMAKE_CURRENT_BINARY_DIR}/gen)
set(gen_dst ${gen_loc}/cld_3/protos)

if (DESKTOP_APP_USE_PACKAGED OR LINUX)
    find_package(protobuf CONFIG)
    if (NOT protobuf_FOUND)
        find_package(Protobuf MODULE REQUIRED)
    endif()
    set(protoc_executable protobuf::protoc)
    set(protobuf_lib protobuf::libprotobuf-lite)
elseif (WIN32)
    set(build_loc ${libs_loc}/protobuf/build/$<IF:$<CONFIG:Debug>,Debug,Release>)
    set(protoc_executable ${build_loc}/protoc.exe)
    set(protobuf_lib ${build_loc}/libprotobuf-lite.lib)
else()
    set(protoc_executable ${libs_loc}/protobuf/build/protoc)
    set(protobuf_lib ${libs_loc}/protobuf/build/libprotobuf-lite.a)
endif()
if (PROTOBUF_PROTOC_EXECUTABLE)
    set(protoc_executable ${PROTOBUF_PROTOC_EXECUTABLE})
endif()

include(generate_protobuf.cmake)

function(generate protobuf_name)
    generate_single_protobuf(
        external_cld3_bundled
        ${gen_dst}
        ${protobuf_name}
        ${protoc_executable})
endfunction()

generate(feature_extractor.proto)
generate(sentence.proto)
generate(task_spec.proto)

nice_target_sources(external_cld3_bundled ${cld3_src}
PRIVATE
    feature_extractor.proto
    sentence.proto
    task_spec.proto

    base.cc
    embedding_feature_extractor.cc
    embedding_network.cc
    feature_extractor.cc
    feature_extractor.h
    feature_types.cc
    fml_parser.cc
    language_identifier_features.cc
    lang_id_nn_params.cc
    nnet_language_identifier.cc
    registry.cc
    relevant_script_feature.cc
    sentence_features.cc
    task_context.cc
    task_context_params.cc
    unicodetext.cc
    utils.cc
    workspace.cc

    script_span/generated_entities.cc
    script_span/getonescriptspan.cc
    script_span/getonescriptspan.h
    script_span/utf8statetable.cc
    script_span/offsetmap.cc
    script_span/text_processing.cc
    script_span/text_processing.h
    script_span/fixunicodevalue.cc
)

target_compile_definitions(external_cld3_bundled
PUBLIC
    $<$<NOT:$<CONFIG:Debug>>:NDEBUG>
)

target_compile_options_if_exists(external_cld3_bundled
PRIVATE
    -Wno-implicit-fallthrough
)

target_include_directories(external_cld3_bundled
PUBLIC
    ${cld3_src}
    ${gen_loc}
)

if (NOT DESKTOP_APP_USE_PACKAGED AND NOT LINUX)
    target_include_directories(external_cld3_bundled
    PUBLIC
        ${libs_loc}/protobuf/src
        ${libs_loc}/protobuf/third_party/abseil-cpp
    )
endif()

target_link_libraries(external_cld3_bundled
PRIVATE
    ${protobuf_lib}
)

target_link_libraries(external_cld3
INTERFACE
    external_cld3_bundled
)
