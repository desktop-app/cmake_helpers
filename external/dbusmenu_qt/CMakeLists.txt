# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

if (DESKTOP_APP_USE_PACKAGED AND NOT DESKTOP_APP_USE_PACKAGED_LAZY)
    add_library(external_dbusmenu_qt INTERFACE IMPORTED GLOBAL)
    add_library(desktop-app::external_dbusmenu_qt ALIAS external_dbusmenu_qt)

    find_package(dbusmenu-qt5 REQUIRED)
    target_link_libraries(external_dbusmenu_qt INTERFACE dbusmenu-qt5)
else()
    add_library(external_dbusmenu_qt OBJECT)
    add_library(desktop-app::external_dbusmenu_qt ALIAS external_dbusmenu_qt)
    init_target(external_dbusmenu_qt "(external)")

    set(dbusmenu_qt_VERSION_MAJOR 0)
    set(dbusmenu_qt_VERSION_MINOR 9)
    set(dbusmenu_qt_VERSION_PATCH 2)

    set(dbusmenu_qt_loc ${third_party_loc}/libdbusmenu-qt/src)

    set_target_properties(external_dbusmenu_qt PROPERTIES AUTOMOC ON)

    # QIcon::name was introduced in Qt 4.7.
    set(HAVE_QICON_NAME ON)

    configure_file(${dbusmenu_qt_loc}/dbusmenu_config.h.in "${CMAKE_CURRENT_BINARY_DIR}/dbusmenu_config.h" @ONLY)
    configure_file(${dbusmenu_qt_loc}/dbusmenu_version.h.in "${CMAKE_CURRENT_BINARY_DIR}/dbusmenu_version.h")

    nice_target_sources(external_dbusmenu_qt ${dbusmenu_qt_loc}
    PRIVATE
        dbusmenu_p.cpp
        dbusmenuexporter.cpp
        dbusmenuexporterdbus_p.cpp
        dbusmenuimporter.cpp
        dbusmenutypes_p.cpp
        dbusmenushortcut_p.cpp
        utils.cpp
    )

    qt5_add_dbus_adaptor(dbusmenu_adaptor
        ${dbusmenu_qt_loc}/com.canonical.dbusmenu.xml
        ${dbusmenu_qt_loc}/dbusmenuexporterdbus_p.h
        DBusMenuExporterDBus
    )

    target_sources(external_dbusmenu_qt PRIVATE ${dbusmenu_adaptor})

    target_include_directories(external_dbusmenu_qt
    PUBLIC
        ${dbusmenu_qt_loc}
        "${CMAKE_CURRENT_BINARY_DIR}"
    )

    target_link_libraries(external_dbusmenu_qt
    PUBLIC
        desktop-app::external_qt
    )
endif()
