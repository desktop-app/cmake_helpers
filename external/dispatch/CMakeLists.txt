# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

set(dispatch_loc ${third_party_loc}/dispatch)
set(dispatch_prefix ${CMAKE_CURRENT_BINARY_DIR}/dispatch-prefix)
set(dispatch_binary_dir ${dispatch_prefix}/src/dispatch-build)

if (DESKTOP_APP_USE_PACKAGED)
    find_library(DESKTOP_APP_DISPATCH_LIBRARIES dispatch)
    find_path(DESKTOP_APP_DISPATCH_INCLUDE_DIRS dispatch/dispatch.h)
endif()

if (NOT DESKTOP_APP_DISPATCH_LIBRARIES OR NOT DESKTOP_APP_DISPATCH_INCLUDE_DIRS)
    execute_process(
        COMMAND ${CMAKE_COMMAND} ${dispatch_loc}
            -B ${dispatch_binary_dir}
            -GNinja
            -DCMAKE_BUILD_TYPE=Release
            -DCMAKE_C_COMPILER=clang
            -DCMAKE_CXX_COMPILER=clang++
            -DCMAKE_C_FLAGS=
            -DCMAKE_CXX_FLAGS=
            -DBUILD_SHARED_LIBS=OFF
            -DBUILD_TESTING=OFF
    )

    execute_process(
        COMMAND ${CMAKE_COMMAND} --build ${dispatch_binary_dir} --parallel
    )

    if (NOT EXISTS ${dispatch_binary_dir}/src/libdispatch.a)
        if (DESKTOP_APP_SPECIAL_TARGET)
            message(FATAL_ERROR "Dispatch build failed")
        else()
            return()
        endif()
    endif()
endif()

add_library(external_dispatch INTERFACE IMPORTED GLOBAL)
add_library(desktop-app::external_dispatch ALIAS external_dispatch)

if (DESKTOP_APP_USE_PACKAGED AND DESKTOP_APP_DISPATCH_LIBRARIES AND DESKTOP_APP_DISPATCH_INCLUDE_DIRS)
    target_include_directories(external_dispatch SYSTEM INTERFACE ${DESKTOP_APP_DISPATCH_INCLUDE_DIRS})
    target_link_libraries(external_dispatch INTERFACE ${DESKTOP_APP_DISPATCH_LIBRARIES})
    return()
endif()

add_library(external_dispatch_bundled STATIC IMPORTED)

set_target_properties(external_dispatch_bundled PROPERTIES
    IMPORTED_LOCATION "${dispatch_binary_dir}/src/libdispatch.a"
)

target_include_directories(external_dispatch_bundled SYSTEM
INTERFACE
    ${dispatch_loc}
)

target_link_libraries(external_dispatch_bundled
INTERFACE
    ${dispatch_binary_dir}/src/BlocksRuntime/libBlocksRuntime.a
)

target_link_libraries(external_dispatch
INTERFACE
    external_dispatch_bundled
)
