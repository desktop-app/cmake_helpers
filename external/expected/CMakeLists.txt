# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

add_library(external_expected INTERFACE IMPORTED GLOBAL)
add_library(desktop-app::external_expected ALIAS external_expected)

if (DESKTOP_APP_USE_PACKAGED_EXPECTED AND NOT DESKTOP_APP_USE_PACKAGED_LAZY)
    find_package(tl-expected CONFIG REQUIRED)
    target_link_libraries(external_expected INTERFACE tl::expected)
else()
    target_include_directories(external_expected SYSTEM
    INTERFACE
        ${third_party_loc}/expected/include
    )
endif()
